MSAL/test/app/mac/MSALAcquireTokenViewController.m

```diff
@@ -0,0 +1,386 @@
+        return MSALPromptTypeSelectAccount;
+    if ([promptType isEqualToString:@"Login"])
+        return MSALPromptTypeLogin;
+    if ([promptType isEqualToString:@"Consent"])
+        return MSALPromptTypeConsent;
+    
+    @throw @"Do not recognize prompt behavior";
+}
+
+- (BOOL)passedInWebview
+{
+    NSString* webViewType = [self.webViewType labelForSegment:[self.webViewType selectedSegment]];
+    
+    if ([webViewType isEqualToString:@"MSAL"])
+    {
+        return NO;
+    }
+    else if ([webViewType isEqualToString:@"Passed In"])
+    {
+        return YES;
+    }
+    else
+    {
+        @throw @"unexpected webview type";
+    }
+}
+
+- (void)showAlert:(NSString *)messageText informativeText:(NSString *)informativeText
+{
+    dispatch_async(dispatch_get_main_queue(), ^{
+        NSAlert *alert = [[NSAlert alloc] init];
+        alert.messageText = messageText;
+        [alert addButtonWithTitle:@"OK"];
+        alert.informativeText = informativeText;
+        [alert runModal];
+    });
+}
+
+- (IBAction)clearCache:(id)sender
+{
+    MSALTestAppSettings *settings = [MSALTestAppSettings settings];
+    
+    // Delete accounts.
+    NSDictionary *currentProfile = [MSALTestAppSettings currentProfile];
+    NSString *clientId = [currentProfile objectForKey:MSAL_APP_CLIENT_ID];
+    NSString *redirectUri = [currentProfile objectForKey:MSAL_APP_REDIRECT_URI];
+    __auto_type authority = [settings authority];
+    
+    NSError *error = nil;
+    MSALPublicClientApplicationConfig *pcaConfig = [[MSALPublicClientApplicationConfig alloc] initWithClientId:clientId
+                                                                                                   redirectUri:redirectUri
+                                                                                                     authority:authority];
+    if (self.validateAuthority.selectedSegment == 1)
+    {
+        pcaConfig.knownAuthorities = @[pcaConfig.authority];
+    }
+    
+    MSALPublicClientApplication *application = [[MSALPublicClientApplication alloc] initWithConfiguration:pcaConfig error:&error];
+    
+    if (!application)
+    {
+        NSString *resultText = [NSString stringWithFormat:@"Failed to create PublicClientApplication:\n%@", error];
+        [self.resultView setString:resultText];
+        return;
+    }
+    
+    BOOL result = [application.tokenCache clearWithContext:nil error:&error];
+    
+    if (result)
+    {
+        [self.resultView setString:@"Successfully cleared cache."];
+        settings.currentAccount = nil;
+        
+        [[NSNotificationCenter defaultCenter] postNotificationName:MSALTestAppCacheChangeNotification object:self];
+    }
+    else
+    {
+        [self.resultView setString:[NSString stringWithFormat:@"Failed to clear cache, error = %@", error]];
+    }
+}
+
+- (IBAction)clearCookies:(id)sender
+{
+    NSHTTPCookieStorage* cookieStore = [NSHTTPCookieStorage sharedHTTPCookieStorage];
```

Comment:
If we are only using WKWebView, do we need this clearing here?

Reply:
Agree. Removed it.